src/ferc_xbrl_extractor/xbrl.py

```diff
@@ -252,30 +252,24 @@ def get_fact_tables(
 
                 taxonomy_entry_point = f"taxonomy/form{form_number}/{taxonomy_date}/form/form{form_number}/form-{form_number}_{taxonomy_date}.xsd"
                 taxonomy = Taxonomy.from_source(f, entry_point=taxonomy_entry_point)
-                taxonomies.append(taxonomy)
+                taxonomies[taxonomy_version] = taxonomy
 
-            datapackage = Datapackage.from_taxonomy(
-                taxonomy, db_uri, form_number=form_number
-            )
+    datapackage = Datapackage.from_taxonomies(
+        taxonomies, db_uri, form_number=form_number
+    )
 
-            if datapackage_path:
-                # Verify that datapackage descriptor is valid before outputting
-                report = Package.validate_descriptor(
-                    datapackage.model_dump(by_alias=True)
-                )
+    if datapackage_path:
+        # Verify that datapackage descriptor is valid before outputting
+        report = Package.validate_descriptor(datapackage.model_dump(by_alias=True))
 
-                if not report.valid:
-                    raise RuntimeError(
-                        f"Generated datapackage is invalid - {report.errors}"
-                    )
+        if not report.valid:
+            raise RuntimeError(f"Generated datapackage is invalid - {report.errors}")
 
-                # Write to JSON file
-                with Path(datapackage_path).open(mode="w") as f:
-                    f.write(datapackage.model_dump_json(by_alias=True))
+        # Write to JSON file
+        with Path(datapackage_path).open(mode="w") as f:
+            f.write(datapackage.model_dump_json(by_alias=True))
```

Comment:
Elsewhere we use `indent=n` to make our datapackages far more readable. Maybe worth considering here as the datapackage is currently hard to read.

Reply:
See e.g., in our archiver repo:

new_datapackage.model_dump_json(by_alias=True, indent=4),
                    encoding="utf-8",

Reply:
Good call, fixed!